spec/rspec/core/bisect/runner_spec.rb

```diff
@@ -213,6 +213,27 @@ def repro_command_from(ids)
       open3_method = Open3.respond_to?(:capture2e) ? :capture2e : :popen3
       open3_method = :popen3 if RSpec::Support::Ruby.jruby?
 
+      RSpec::Matchers.define :invoke_command_with_env do |command, environment|
+        match do |actual|
+          if open3_method == :capture2e
+            actual.call
+            expect(Open3).to have_received(open3_method).with(environment, command)
+          elsif open3_method == :popen3
+            called_environment = nil
+            allow(Open3).to receive(open3_method) do
+              called_environment = ENV.to_hash.dup
+            end
```

Comment:
It's a little weird that you stub this here but not for the `capture2e` branch.  In the `before` hook below I see that this method is also stubbed, but then I get to wondering how these two stubs interact. Is there a way we can clean it up and make it more consistent?